modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/JSResolveModulesServlet.java

```diff
@@ -44,22 +43,13 @@
 		"osgi.http.whiteboard.servlet.pattern=/js_resolve_modules",
 		"service.ranking:Integer=" + Details.MAX_VALUE_LESS_1K
 	},
-	service = {
-		JSResolveModulesServlet.class, NPMRegistryUpdatesListener.class,
-		Servlet.class
-	}
+	service = Servlet.class
 )
-public class JSResolveModulesServlet
-	extends HttpServlet implements NPMRegistryUpdatesListener {
+public class JSResolveModulesServlet extends HttpServlet {
 
-	public JSResolveModulesServlet() {
-		onAfterUpdate();
-	}
-
-	@Override
-	public void onAfterUpdate() {
-		_etag = StringBundler.concat(
-			"W/\"", UUID.randomUUID(), StringPool.QUOTE);
+	@Activate
+	protected void activate() {
+		JSLoaderModulesUtil.updateJSLoaderProps();
```

Comment:
@MrJohn1911 this should not be needed, you can make JSLoaderModulesUtil initialize its field by default.